ouf.lua

```diff
@@ -564,6 +564,98 @@ function oUF:Spawn(unit, overrideName)
 	return object
 end
 
+function oUF:SpawnNamePlates(namePrefix, nameplateCallback, nameplateCVars)
+	argcheck(nameplateCallback, 3, 'function', 'nil')
+	argcheck(nameplateCVars, 4, 'table', 'nil')
+
+	local style = style
+	if(not style) then return error('Unable to create frame. No styles have been registered.') end
+
+	local prefix = namePrefix or generateName()
+
+	local hiddenParent = CreateFrame('Frame', nil, UIParent)
+	hiddenParent:SetAllPoints()
+	hiddenParent:Hide()
+
+	NamePlateDriverFrame:Hide()
+	NamePlateDriverFrame:UnregisterAllEvents()
+	NamePlateDriverFrame:RegisterEvent('NAME_PLATE_CREATED')
+	NamePlateDriverFrame:RegisterEvent('NAME_PLATE_UNIT_ADDED')
+	NamePlateDriverFrame:RegisterEvent('NAME_PLATE_UNIT_REMOVED')
+	NamePlateDriverFrame:HookScript('OnEvent', function(_, event, unit)
+		if(event == 'NAME_PLATE_UNIT_ADDED') then
+			if(not unit) then return end
+
+			local nameplate = C_NamePlate.GetNamePlateForUnit(unit)
+
+			if(nameplate.UnitFrame) then
+				nameplate.UnitFrame:SetParent(hiddenParent)
```

Comment:
I'm assuming that the sub-elements on this frame have events registered, might be a good idea to unregister them.